### DISCUSSION TITLE
about concerns for using quasar umd

### DISCUSSION THREAD

**Discussion by bobxmas** [quasarframework/quasar#16061] [General - Components / Directives / etc]
Dear Team,
I have an existing vue 2 project and would like to use quasar components with customization.
I've seached for quasar and realized that general method is first, make a project with quasar CLI then migrate my project.
But my project uses a framework made by my team,  so migration is risky. I saw qausar umd and you said it can be used for 
progressive migration. I did PoC for it, code cdn link and import QBtn at some .vue file. it works.
But I have concerns. 
1) There are 2 vue instances: 
  - one is from node module with package.json, which is set with project options and provide environment for project.
  - another is from cdn, which is for quasar. without it, quasar didn't make Quasar instance as a global variable.
  concern 1. I am worry, is there some risk? like conflict. 
  I think, two vues are serparated with different usages(for project and quasar component), so, it would be safe. I hope so.
 But I need your thoughts (because I don't know the codes of quasar and vue.)
2) When customizing quasar component, is there some concerns? I don't know how to customize quasar component yet.

Please give me your tips and thoughts.

**Comment by suleimanas** [comment]
https://v1.quasar.dev/start/umd

**Reply by bobxmas** [reply]
I have already read the guide for umd and set it at my project.
But there are some concerns, so I posted my concerns at here.

**Reply by suleimanas** [reply]
without telling us more about your first concern I can only send you the guide.
but I can answer your concern number 2 there is no issues. you can customizing quasar components (it's vue components after all) on any quasar component guide you would find (an API section) in it: (Props + Slots + Events + Methods): https://v1.quasar.dev/vue-components/button#qbtn-api
